google-cloud-storage/src/test/java/com/google/cloud/storage/it/StorageNativeCanary.java

```diff
@@ -0,0 +1,182 @@
+   *   <li>Insert two (2) objects (Unary, ServerStream)
+   *   <li>List all objects, using a pageSize of 1 (Page over Unary)
+   *   <li>Read all bytes of each object (ServerStream)
+   *   <li>Delete each object (Unary)
+   *   <li>Delete temporary bucket (Unary)
+   * </ul>
+   */
+  private static void canary_happyPath(Storage storage) throws Exception {
+    // create a temporary bucket
+    try (TemporaryBucket tempB =
+        TemporaryBucket.newBuilder()
+            .setStorage(storage)
+            .setBucketInfo(BucketInfo.of("java-storage-grpc-" + UUID.randomUUID()))
+            .build()) {
+      String bucketName = tempB.getBucket().getName();
+      String obj1Name = UUID.randomUUID().toString();
+      String obj2Name = UUID.randomUUID().toString();
+
+      // insert 2 objects
+      BlobInfo info1 = BlobInfo.newBuilder(bucketName, obj1Name).build();
+      BlobInfo info2 = BlobInfo.newBuilder(bucketName, obj2Name).build();
+      uploadUsingWriter(storage, info1);
+      uploadUsingWriter(storage, info2);
+
+      // list objects
+      Page<Blob> page = storage.list(bucketName, BlobListOption.pageSize(1));
+      List<Blob> blobs = ImmutableList.copyOf(page.iterateAll());
+
+      // read all bytes of each object
+      List<BlobWithContent> actual =
+          blobs.stream()
+              .map(info -> readAll(storage, info))
+              .collect(ImmutableList.toImmutableList());
+
+      List<Boolean> deletes =
+          blobs.stream()
+              .map(b -> storage.delete(b.getBlobId(), BlobSourceOption.generationMatch()))
+              .collect(ImmutableList.toImmutableList());
+
+      assertAll(
+          () -> {
+            List<String> actualNames =
+                actual.stream()
+                    .map(BlobWithContent::getInfo)
+                    .map(BlobInfo::getBlobId)
+                    .map(BlobId::getName)
+                    .collect(ImmutableList.toImmutableList());
+
+            assertThat(actualNames).containsExactly(info1.getName(), info2.getName());
+          },
+          () -> assertThat(actual.get(0).getContent()).isEqualTo(bytes),
+          () -> assertThat(actual.get(1).getContent()).isEqualTo(bytes),
+          () -> assertThat(deletes.get(0)).isTrue(),
+          () -> assertThat(deletes.get(1)).isTrue());
+    }
+  }
+
+  private static void uploadUsingWriter(Storage storage, BlobInfo info) throws IOException {
+    try (WriteChannel w = storage.writer(info, BlobWriteOption.doesNotExist())) {
+      // set our size to the smallest resumable size, so we can send multiple requests
+      w.setChunkSize(_256KiB);
+      ByteStreams.copy(Channels.newChannel(new ByteArrayInputStream(bytes)), w);
+    }
+  }
+
+  private static BlobWithContent readAll(Storage storage, BlobInfo info) {
+    try (ReadChannel r = storage.reader(info.getBlobId(), BlobSourceOption.generationMatch());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
optional: how about this? 
```suggestion
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```